### PR TITLE
VxCentralScan: improve `Importer` correctness

### PR DESCRIPTION
**Pull request by eventualbuddha**
## Overview

Refs #8041 

Before making structural changes to VxCentralScan to support interpreting while scanning, I wanted to improve the foundation a bit. This PR fixes a few issues Claude and I found while perusing `Importer`. Mostly they're edge case handlng, but could represent real correctness issues.

## Demo Video or Screenshot
n/a

## Testing Plan
Existing automated tests.

### CODE REVIEW

File: apps/central-scan/backend/src/importer.ts [hunk 1]
```diff
@@ -305,7 +307,6 @@ export class Importer {
     await logBatchComplete(this.logger, batch);
```
[the remark this answers is not in the document]
Comment (eventualbuddha) [reply apps/central-scan/backend/src/importer.ts]: Agreed, will incorporate.

File: apps/central-scan/backend/src/importer.ts [hunk 2]
```diff
@@ -335,51 +336,82 @@ export class Importer {
...
+        if (typeof batchScanDirectory !== 'undefined') {
+          await fsExtra.remove(batchScanDirectory);
+        }
+      } else {
+        await this.finishBatch(extractErrorMessage(error));
+      }
+      throw error;
+    } finally {
+      this.isStartingBatch = false;
+    }
```
[the remark this answers is not in the document]
Comment (eventualbuddha) [reply apps/central-scan/backend/src/importer.ts]: It's complex enough that I think a test is warranted.

File: apps/central-scan/backend/src/importer.ts [hunk 3]
```diff
@@ -296,16 +297,18 @@ export class Importer {
     if (!currentBatch) {
       return;
     }
+    this.currentBatch = undefined;
```
Comment (kshen0) [inline apps/central-scan/backend/src/importer.ts]: Should we add a test for re-entry cleanup?

**Review by kshen0 — APPROVED** [review]
